### PR TITLE
Clean Up src/parsers

### PR DESCRIPTION
**Pull request by WillAyd**
Looking at this in detail noticed quite a few unused functions and pre-processor directives. There was also a mismatch in types between the .pyx and C structs

### CODE REVIEW

File: pandas/_libs/src/parser/tokenizer.h [hunk 1]
```diff
@@ -191,7 +158,7 @@ typedef struct parser_t {
     void *skipset;
     PyObject *skipfunc;
     int64_t skip_first_N_rows;
-    int skip_footer;
+    int64_t skip_footer;
```
Comment (WillAyd) [inline pandas/_libs/src/parser/tokenizer.h]: Changed to match definition in parsers.pyx

File: pandas/_libs/src/parser/tokenizer.h [hunk 2]
```diff
@@ -208,7 +175,7 @@ typedef struct parser_t {
 typedef struct coliter_t {
     char **words;
     int64_t *line_start;
-    int col;
+    int64_t col;
```
Comment (WillAyd) [inline pandas/_libs/src/parser/tokenizer.h]: Ditto

File: pandas/_libs/src/parser/tokenizer.h [hunk 3]
```diff
@@ -31,9 +25,6 @@ See LICENSE for the license
 
 #include "khash.h"
 
-#define CHUNKSIZE 1024 * 256
```
Comment (jreback) [inline pandas/_libs/src/parser/tokenizer.h]: these are not used?

Comment (WillAyd) [reply pandas/_libs/src/parser/tokenizer.h]: No; there is a `DEFAULT_CHUNKSIZE`defined in parsers.pyx which appears used instead

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by WillAyd** [comment]
@jreback all green

**Comment by jreback** [comment]
thanks

**Pull request merged**